app/src/main/java/com/pokedex/koto/pokedex/MainActivity.java

```diff
@@ -1,51 +1,54 @@
 package com.pokedex.koto.pokedex;
 
-import android.os.AsyncTask;
-import android.support.v7.app.AppCompatActivity;
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+import retrofit2.Retrofit;
+import retrofit2.converter.gson.GsonConverterFactory;
+
 import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
 import android.util.Log;
+import android.widget.Toast;
 
-import java.io.IOException;
 import java.util.List;
 
-import retrofit2.Call;
-import retrofit2.Retrofit;
-import retrofit2.converter.gson.GsonConverterFactory;
-
 public class MainActivity extends AppCompatActivity {
 
+    private String url = "http://pokeapi.co/api/v2/";
+    private Retrofit retrofit;
+    private PokemonService service;
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
 
-        new Peticion().execute();
-    }
-
-    public static class Peticion extends AsyncTask<Void,Void,Void>{
-        @Override
-        protected Void doInBackground(Void... params) {
-            final String url = "http://pokeapi.co/api/v2/";
-            Retrofit retrofit = new Retrofit.Builder()
-                    .baseUrl(url)
-                    .addConverterFactory(GsonConverterFactory.create())
-                    .build();
+        //Init Retrofit
+        retrofit = new Retrofit.Builder()
+                .baseUrl(url)
+                .addConverterFactory(GsonConverterFactory.create())
+                .build();
 
-            servicePokemon service = retrofit.create(servicePokemon.class);
+        service = retrofit.create(PokemonService.class);
+        service.listPokemon().enqueue(new Callback<ListResponse>() {
```

Comment:
Retrofit already offers an asynchronous way of doing API calls, this way we don't need to use `AsyncTask`, so I simplified it removing the `AsyncTask`. Even though the `AsyncTask` way was not incorrect.